libs/feature/record/src/lib/map-view/map-view.component.ts

```diff
@@ -96,36 +100,38 @@ export class MapViewComponent implements AfterViewInit {
   selection: Feature
   showLegend = true
   legendExists = false
+  loading = false
+  error = null
+
+  selectLinkToDisplay(i: number) {
+    this.selectedLinkIndex$.next(i)
+  }
+
+  selectStyleToDisplay(i: number) {
+    this.selectedStyleIndex$.next(i)
+  }
 
   toggleLegend() {
     this.showLegend = !this.showLegend
   }
-
-  onLegendStatusChange(status: boolean) {
-    this.legendExists = status
+  onLegendStatusChange(v: boolean) {
+    this.legendExists = v
   }
 
   compatibleMapLinks$ = combineLatest([
     this.mdViewFacade.mapApiLinks$,
     this.mdViewFacade.geoDataLinksWithGeometry$,
   ]).pipe(
-    switchMap(async ([mapApiLinks, geoDataLinksWithGeometry]) => {
-      // looking for TMS links to process
-      let processedMapApiLinks = await Promise.all(
-        mapApiLinks.map((link) => {
-          if (link.type === 'service' && link.accessServiceProtocol === 'tms') {
-            return this.dataService.getGeodataLinksFromTms(link)
-          }
-          return link
-        })
-      )
-      processedMapApiLinks = processedMapApiLinks.flat()
-      return [...processedMapApiLinks, ...geoDataLinksWithGeometry]
-    }),
+    map(([mapApiLinks, geoDataLinksWithGeometry]) => [
+      ...mapApiLinks,
+      ...geoDataLinksWithGeometry,
+    ]),
     shareReplay(1)
   )
 
-  dropdownChoices$ = this.compatibleMapLinks$.pipe(
+  sourceLinks$ = this.compatibleMapLinks$.pipe(shareReplay(1))
```

Comment:
`sourceLinks$` and `compatibleMapLinks$` are the same, so I think you can get rid of one :-)